src/operators/switch.ml

```diff
@@ -123,6 +123,11 @@ object (self)
 
   method is_ready = need_eot || selected <> None || self#cached_select <> None
 
+  method self_sync =
```

Comment:
I'm not sure I understand this one...

Reply:
If there's a selected source return its `self_sync` otherwise say `false`? In the last case, I believe the source will not be `is_ready` so it should never be pulled anyway.